### PR TITLE
Hero entity and DAO

### PR DESCRIPTION
**Pull request by xsestak4**

### CODE REVIEW

**Comment by VDuchon** [comment]
Testing hero entity (in progress).

**Comment by xtracko** [comment]
@Vduchon I have no idea how to fix it, or even what is specifically broken. Maybe @MiroslavMacor could help.

**Comment by xtracko** [comment]
@VDuchon @MiroslavMacor Maybe I figured it out. You must merge the branch troop-entity to hero-entity because Hero has ManyToOne relationship with Troop, but no definition of Troop is provided.

Edit: Try to merge branch role-entity too for the same reason.

File: dndtroops-persistence/src/main/java/cz/muni/fi/pa165/dndtroops/dao/HeroDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    @Override
+    public Hero findById(Long id) {
+       return em.find(Hero.class, id);
+    }
+
+    @Override
+    public Hero findByName(String name) {
+       try {
+            return em.createQuery("select h from Hero h where h.name = :name", Hero.class)
```
Comment (MiroslavMacor) [inline dndtroops-persistence/src/main/java/cz/muni/fi/pa165/dndtroops/dao/HeroDaoImpl.java]: SELECT FROM WHERE should be uppercase

File: dndtroops-persistence/src/main/java/cz/muni/fi/pa165/dndtroops/dao/HeroDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+                .getSingleResult();
+         } catch (NoResultException ex) {
+            return null;
+         }
+    }
+
+    @Override
+    public List<Hero> findByRole(Role role) {
+        try {
+            return em.createQuery("select h from Hero h where h.role = :roleid", Hero.class)
```
Comment (MiroslavMacor) [inline dndtroops-persistence/src/main/java/cz/muni/fi/pa165/dndtroops/dao/HeroDaoImpl.java]: Same here and rest of the class

**Pull request merged**